src/parcsr_ls/par_amg_setup.c

```diff
@@ -1450,7 +1446,6 @@ hypre_BoomerAMGSetup( void               *amg_vdata,
                if (AN) hypre_ParCSRMatrixDestroy(AN);
                if (num_functions > 1) hypre_TFree(coarse_dof_func, HYPRE_MEMORY_HOST);
                hypre_TFree(CF_marker, HYPRE_MEMORY_HOST);
-               hypre_TFree(coarse_pnts_global, HYPRE_MEMORY_HOST);
```

Comment:
I moved this to `par_coarse_parms.c`: https://github.com/hypre-space/hypre/blob/dd34cf278bffcfb2f1f896a31c3e0bd83e4d547e/src/parcsr_ls/par_coarse_parms.c#L118

so that we don't need to remember freeing `coarse_pnts_global` in `BoomerAMGSetup` when it isn't needed anymore (in the original code, this was giving me some memleaks difficult to track).

Actually, I was thinking about your suggestion of moving `partitioning` to the stack. The same could be done here for `coarse_pnts_global`. What do you think?

Reply:
OK. I see how it works now, a bit tricky though. But it is fine. Thanks!